specs/DHT.md

```diff
@@ -0,0 +1,399 @@
+# libp2p Daemon Control Protocol
+
+The libp2p daemon DHT protocol allows clients to query and announce to the
+libp2p DHT.
+
+_At the moment, this is a living document. As such, it will be susceptible to
+changes until stabilization._
+
+## Protocol Specification
+
+### Data Types
+
+The data structures are defined in [pb/p2pd.proto](pb/p2pd.proto). All messages
+are varint-delimited. For the DHT queries, the relevant data types are:
+
+- `DHTRequest`
+- `DHTResponse`
+
+All DHT requests will be wrapped in a `Request` message with `Type: DHT`. All
+DHT responses from the daemon will be wrapped in a `Response` with the
+`DHTResponse` field populated.
```

Comment:
actually, not all responses do that -- there are things that don't return a useful value other than the error indication (eg `Provide`).